server/src/services/checkSubmission.js

```diff
@@ -0,0 +1,174 @@
+class checkSubmission {
+  constructor(sudokuPuzzle){
+    this.sudokuPuzzle = sudokuPuzzle
+  }
+
+  checkDuplicates(array) {
+    const containDuplicates = array.some(
+      (val, i) => array.indexOf(val) !== i
+    )
+    return containDuplicates
+  }
+
+  checkRow() {
+    let row1 = []
+    let row2 = []
+    let row3 = []
+    let row4 = []
+    let row5 = []
+    let row6 = []
+    let row7 = []
+    let row8 = []
+    let row9 = []
+
+    for(let x = 0; x < 3; x++) {
+      row1.push(this.sudokuPuzzle.box1[x])
+      row2.push(this.sudokuPuzzle.box1[x+3])
+      row3.push(this.sudokuPuzzle.box1[x+6])
+      row4.push(this.sudokuPuzzle.box4[x])
+      row5.push(this.sudokuPuzzle.box4[x+3])
+      row6.push(this.sudokuPuzzle.box4[x+6])
+      row7.push(this.sudokuPuzzle.box7[x])
+      row8.push(this.sudokuPuzzle.box7[x+3])
+      row9.push(this.sudokuPuzzle.box7[x+6])
+    }
+
+    for(let x = 0; x < 3; x++) {
+      row1.push(this.sudokuPuzzle.box2[x])
+      row2.push(this.sudokuPuzzle.box2[x+3])
+      row3.push(this.sudokuPuzzle.box2[x+6])
+      row4.push(this.sudokuPuzzle.box5[x])
+      row5.push(this.sudokuPuzzle.box5[x+3])
+      row6.push(this.sudokuPuzzle.box5[x+6])
+      row7.push(this.sudokuPuzzle.box8[x])
+      row8.push(this.sudokuPuzzle.box8[x+3])
+      row9.push(this.sudokuPuzzle.box8[x+6])
+    }
+
+    for(let x = 0; x < 3; x++) {
+      row1.push(this.sudokuPuzzle.box3[x])
+      row2.push(this.sudokuPuzzle.box3[x+3])
+      row3.push(this.sudokuPuzzle.box3[x+6])
+      row4.push(this.sudokuPuzzle.box6[x])
+      row5.push(this.sudokuPuzzle.box6[x+3])
+      row6.push(this.sudokuPuzzle.box6[x+6])
+      row7.push(this.sudokuPuzzle.box9[x])
+      row8.push(this.sudokuPuzzle.box9[x+3])
+      row9.push(this.sudokuPuzzle.box9[x+6])
+    }
+
+    if(this.checkDuplicates(row1) &&
+    this.checkDuplicates(row2) &&
+    this.checkDuplicates(row3) &&
+    this.checkDuplicates(row4) &&
+    this.checkDuplicates(row5) &&
+    this.checkDuplicates(row6) &&
+    this.checkDuplicates(row7) &&
+    this.checkDuplicates(row8) &&
+    this.checkDuplicates(row9)
+    ){
+      return true
+    } else {
+      return false
+    }
+
+
+  }
```

Comment:
indentation and spacing! We can also refactor this to fewer lines with a loop! 
In the future (though not necessarily right now) you can save WHICH of the rows has a duplicate